protocols/identify/src/lib.rs

```diff
@@ -99,6 +105,80 @@ impl<T: Callback> IdentifyProtocol<T> {
             secio_enabled: true,
         }
     }
+
+    fn process_listens(&mut self, context: &mut ProtocolContextMutRef, listens: Vec<Multiaddr>) {
```

Comment:
Same issue here, suggest to use &[Multiaddr]

Reply:
Here, I want to consume it, so use value. others will change to  slice